torchaudio/csrc/ffmpeg/stream_reader/stream_reader.cpp

```diff
@@ -163,12 +163,42 @@ bool StreamReader::is_buffer_ready() const {
 ////////////////////////////////////////////////////////////////////////////////
 // Configure methods
 ////////////////////////////////////////////////////////////////////////////////
-void StreamReader::seek(double timestamp) {
-  TORCH_CHECK(timestamp >= 0, "timestamp must be non-negative.");
+void StreamReader::seek(double timestamp_s, int64_t mode) {
+  TORCH_CHECK(timestamp_s >= 0, "timestamp must be non-negative.");
+  TORCH_CHECK(
+      pFormatContext->nb_streams > 0,
+      "At least one stream must exist in this context");
+
+  int64_t timestamp_av_tb = static_cast<int64_t>(timestamp_s * AV_TIME_BASE);
+  // rescale timestamp to the default stream timebase
+  AVRational default_tb = pFormatContext->streams[0]->time_base;
+  int64_t timestamp_default_tb =
+      av_rescale_q(timestamp_av_tb, AV_TIME_BASE_Q, default_tb);
```

Comment:
I am not sure why this is necessary to compute PTS on the first stream here. Isn't it more straightforward to pass discard target time stamp as second and let each StreamProcessor converts it to pts of the corresponding stream, then let Sink class discard the frame output from FilterGraph by comparing it?

Reply:
This is just for the initial seek (`av_seek_frame`).

Reply:
Why are we manually converting it to the time base of the **first stream** (which is not necessarily default stream)?

https://ffmpeg.org/doxygen/4.1/group__lavf__decoding.html#gaa23f7619d8d4ea0857065d9979c75ac8

According to the doc, `av_seek_frame` selects the default stream and do the conversion automatically with `stream_index=-1`. Is there a case that this is undesired?

Reply:
you are right.